src/main/java/com/api/mapper/RocketMapper.java

```diff
@@ -26,32 +28,64 @@
 import com.api.entity.SxSecondStage;
 import com.api.entity.SxThrustStage;
 import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.springframework.stereotype.Component;
 
-@Mapper
+@Mapper(componentModel = "spring")
 public interface RocketMapper {
     RocketDto toDto(SxRocket sxRocket);
 
+    SxRocket toDto(RocketDto sxRocket);
```

Comment:
same here, why you have SX to DT added? as the flow is Controller -> fetch from Rocket API -> Save to DB -> Get from DB -> DTO response in controller